File: GLES/src/gen_div_table.c
#include <stdio.h>
#include <stdlib.h>
#include "fixed.h"

#define MAX_DIV 2048

int main(int argc, char **argv) {
    int i;
    FILE *f = fopen("div_table.h", "w");

    if (f == NULL) return EXIT_FAILURE;

    fprintf(f, "GLfixed div_table[] = {\n");

    fprintf(f, "%i, ", 0);
    for (i = 1;i < MAX_DIV;i++) {

    	fprintf(f, "%i", fixed_float((1.0f / (float)i)));

    	if (i == (MAX_DIV - 1)) fprintf(f, " \n};\n");
    	else fprintf(f, ", ");
    	if (i % 32 == 0) fprintf(f, "\n");
    }
    fclose(f);
    return EXIT_SUCCESS;
}

File: GLES/src/gen_div_table.py
import math

FIXED_SHIFT = 16

def fixed(d):
    return math.ceil(d * (1 << FIXED_SHIFT))

MAX_DIV = 2048

print("GLfixed div_table[] = {")
print("0, ", end='')

for i in range(1, MAX_DIV):
    if (i + 1) % 16 != 0:
        end = ', '
    else:
        end = ',\n' if i != MAX_DIV - 1 else '\n'

    print(fixed(1 / i), end=end)

print("};")
